great_expectations/experimental/datasources/sqlite_datasource.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Dict, List, Optional, Type
+
+import pydantic
+from pydantic import dataclasses as pydantic_dc
+from typing_extensions import ClassVar
+
+from great_expectations.experimental.datasources.interfaces import DataAsset
+
+if TYPE_CHECKING:
+    import sqlalchemy
+
+from typing_extensions import Literal
+
+from great_expectations.experimental.datasources.sql_datasource import (
+    BatchSortersDefinition,
+    ColumnSplitter,
+    DatetimeRange,
+    SQLDatasource,
+    SQLDatasourceError,
+    TableAsset,
+    _query_for_year_and_month,
+)
+
+
+class SqliteTableAsset(TableAsset):
+    # Subclass overrides
+    type: Literal["sqlite_table"] = "sqlite_table"  # type: ignore[assignment]
+    column_splitter: Optional[SqliteYearMonthSplitter] = None  # type: ignore[assignment]
+
+    # This asset type will support a variety of splitters
+    def add_year_and_month_splitter(
```

Comment:
I made the new `SqliteTableAsset` so I could have this helper function add correct splitter.